src/Orleans.Runtime/Scheduler/OrleansSchedulerAsynchAgent.cs

```diff
@@ -39,10 +39,10 @@ internal class OrleansSchedulerAsynchAgent : AsynchQueueAgent<IWorkItem>
 
         protected override void Process(IWorkItem request)
         {
-            RuntimeContext.InitializeThread(scheduler);
+            RuntimeContext.InitializeThread();
             try
             {
-                RuntimeContext.SetExecutionContext(request.SchedulingContext, scheduler);
+                RuntimeContext.SetExecutionContext(request.SchedulingContext);
```

Comment:
`RuntimeContext.TaskScheduler` was unused. Hopefully it's alright to remove it in this PR, but if not then I can split it out.